test/es-module/test-vm-compile-function-leak.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+// Flags: --max-old-space-size=16 --trace-gc
+
+require('../common');
+const vm = require('vm');
+let count = 0;
+
+function main() {
+  vm.compileFunction(`"${Math.random().toString().repeat(512)}"`, [], {
```

Comment:
```suggestion
  // Math.random().toString() output most of the time a 18-char long string.
  // 512*2048 == 1024**2, so this test is setup to overflows the V8 heap if
  // the generated modules are not garbage collected.
  vm.compileFunction(`"${Math.random().toString().repeat(512)}"`, [], {
```

Reply:
I don't think we need the list the math here because there is always a bunch of overhead involve and the number is picked based on experience (a number that's big enough to reproduce the bug but small enough to not make the test run really slow that it might timeout in the CI). You can't see the correlation between this and the `--max-old-space-size=16` anyway (it's picked because it works locally for me, and frankly I don't think it's possible to have a proper math here due to how variable the overhead can be).